opamp/config.go

```diff
@@ -29,19 +32,82 @@ var (
 
 	// errPrefixParse for error when parsing config
 	errPrefixParse = "failed to parse OpAmp config"
+
+	// errMissingTLSFiles is the error when only one of Key or Cert is specified
+	errMissingTLSFiles = "must specify both Key and Certificate file"
+
+	// errInvalidKeyFile for key file that is not readable
+	errInvalidKeyFile = "failed to read TLS key file"
+
+	// errInvalidCertFile for cert file that is not readable
+	errInvalidCertFile = "failed to read TLS cert file"
+
+	// errInvalidCAFile for ca file that is not readable
+	errInvalidCAFile = "failed to read TLS CA file"
 )
 
 // Config contains the configuration for the collector to communicate with an OpAmp enabled platform.
 type Config struct {
-	Endpoint  string  `yaml:"endpoint"`
-	SecretKey *string `yaml:"secret_key,omitempty"`
-	AgentID   string  `yaml:"agent_id"`
+	Endpoint  string     `yaml:"endpoint"`
+	SecretKey *string    `yaml:"secret_key,omitempty"`
+	AgentID   string     `yaml:"agent_id"`
+	TLS       *TLSConfig `yaml:"tls_config,omitempty"`
 
 	// Updatable fields
 	Labels    *string `yaml:"labels,omitempty"`
 	AgentName *string `yaml:"agent_name,omitempty"`
 }
 
+// TLSConfig represents the TLS config to connect to OpAmp server
+type TLSConfig struct {
+	Insecure bool    `yaml:"insecure"`
+	KeyFile  *string `yaml:"key_file"`
+	CertFile *string `yaml:"cert_file"`
+	CAFile   *string `yaml:"ca_file"`
+}
+
+// ToTLS converts the config to a tls.Config
+func (c Config) ToTLS() (*tls.Config, error) {
+	if c.TLS == nil {
+		return nil, nil
+	}
+
+	tlsConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+	}
+
+	if c.TLS.Insecure {
```

Comment:
looks like the "Insecure" config param is being used synonymously as "InsecureSkipVerify". is there a reason for this? for example in otel's tls config options they have both an "insecure" to turn TLS on/off and a "insecure_skip_verify" to avoid verifying the certificate

Reply:
We could change it.